huaweicloud/services/obs/resource_huaweicloud_obs_bucket_policy.go

```diff
@@ -109,12 +111,14 @@ func resourceObsBucketPolicyRead(_ context.Context, d *schema.ResourceData, meta
 
 	log.Printf("[DEBUG] read policy for obs bucket: %s", d.Id())
 	output, err := obsClient.GetBucketPolicy(d.Id())
-
-	var pol string
-	if err == nil && output.Policy != "" {
-		pol = output.Policy
+	if err != nil {
+		if obsErr, ok := err.(obs.ObsError); ok && obsErr.BaseModel.StatusCode == 404 {
```

Comment:
use `obsErr.StatusCode` will be fine
```suggestion
		if obsErr, ok := err.(obs.ObsError); ok && obsErr.StatusCode == 404 {
```

Reply:
done.